Ical.Net/Serialization/DataTypes/RecurrencePatternSerializer.cs

```diff
@@ -244,7 +244,20 @@ public override object Deserialize(TextReader tr)
                 var keywordPairs = match.Groups[2].Value.Split(';');
                 foreach (var keywordPair in keywordPairs)
                 {
+                    if (keywordPair.Length == 0)
+                    {
+                        // This is illegal but ignore for now.
+                        continue;
+                    }
+
                     var keyValues = keywordPair.Split('=');
+                    if (keyValues.Length != 2)
+                    {
+                        // ArgumentExceptions seem to be the Exception of choise for this class. Should
```

Comment:
ArgumentException is fine for this PR. As we're aware now we could create an issue labelled enhancement to track implementing an Exception type for parsing and/or recurrence failures

Reply:
#648